format_precice_config/format_precice_config.py

```diff
@@ -117,23 +118,192 @@ def printChildren(self, element, level):
                 self.printElement(child, level=level)
             return
 
-        groups1 = itertools.groupby(element.getchildren(),
-                                    lambda e: str(e.tag).split(':')[0])
-
-        groups = []
-
-        for _, group in groups1:
-            group = list(group)
-            if isEmptyTag(group[0]):
-                groups.append(group)
-            else:
-                groups += [[e] for e in group]
-
-        last = len(groups)
-        for i, group in enumerate(groups, start=1):
-            for child in group:
-                self.printElement(child, level=level)
-            if not (isComment(group[0]) or (i == last)):
+        # Custom sorting for top-level elements
+        def custom_sort_key(elem):
+            tag = str(elem.tag)
+            # Predefined order for top-level elements with prefix matching
+            order = {
+                'data:': 1,  # Matches data:vector, data:scalar, etc.
+                'mesh': 2,
+                'participant': 3,
+                'm2n:': 4,
+                'coupling-scheme:': 5
+            }
```

Comment:
Can you define the orders as global constants and write both key functions outside the big loop?

This should make them more compact and call sites will be more readable.